spring-cloud-gcp-samples/spring-cloud-gcp-integration-storage-sample/src/test/java/com/example/GcsSpringIntegrationTests.java

```diff
@@ -104,23 +107,44 @@ void testFilePropagatedToLocalDirectory() {
                   .iterateAll()
                   .forEach(b -> blobNamesInOutputBucket.add(b.getName()));
 
-              assertThat(blobNamesInOutputBucket).contains(TEST_FILE_NAME);
+              assertThat(blobNamesInOutputBucket).contains(TEST_FILE);
             });
   }
 
   void cleanupCloudStorage() {
-    Page<Blob> blobs = this.storage.list(this.cloudInputBucket);
-    for (Blob blob : blobs.iterateAll()) {
-      blob.delete();
+    BlobId inputBucketBlobId = BlobId.of(cloudInputBucket, TEST_FILE);
+    Blob inputBucketBlob = storage.get(inputBucketBlobId);
+    if (inputBucketBlob != null) {
+      inputBucketBlob.delete();
     }
+
+    BlobId outputBucketBlobId = BlobId.of(cloudOutputBucket, TEST_FILE);
+    Blob outputBucketBlob = storage.get(outputBucketBlobId);
+    if (outputBucketBlob != null) {
+      outputBucketBlob.delete();
+    }
+  }
+
+  void cleanupLocalDirectories() throws IOException {
+    cleanupLocalDirectory(Paths.get(uniqueDirectory));
+    cleanupLocalDirectory(Paths.get(outputFolder));
```

Comment:
@burkedavison This implementation was tested by running 4 maven commands in parallel using a simple script that does the following:

```
mvn clean verify &
mvn clean verify &
mvn clean verify &
mvn clean verify &
```
However, wanted to call out a drawback that still exists where the folder name specified through the `${gcs-local-directory}` gets created automatically despite not being used in the tests. The `cleanupLocalDirectory` method is needed to delete this folder at the end of each test run.